tests/integration/components/async-list/component-test.js

```diff
@@ -7,10 +7,10 @@ module("Integration | Component | async list", function (hooks) {
   setupRenderingTest(hooks);
 
   test("yields list on success", async function (assert) {
-    this.set("data", { last: { value: ["a", "b"] } });
+    this.data = { last: { value: ["a", "b"] } };
```

Comment:
We should keep it like it was. Otherwise it will not track those attributes properly.

```
this.set("data", { last: { value: ["a", "b"] } });
```
This is also relevant for all the other occurrences you've changed.
Access this `data` property via `this.data` as you already did.